include/vcpkg/paragraphparser.h

```diff
@@ -75,7 +75,9 @@ namespace vcpkg
     ExpectedL<std::vector<ParsedQualifiedSpecifier>> parse_qualified_specifier_list(const std::string& str,
                                                                                     StringView origin = "<unknown>",
                                                                                     TextRowCol textrowcol = {});
-    ExpectedL<std::vector<Dependency>> parse_dependencies_list(const std::string& str,
-                                                               StringView origin = "<unknown>",
-                                                               TextRowCol textrowcol = {});
+    ExpectedL<std::vector<Dependency>> parse_dependencies_list(
+        const std::string& str,
+        StringView origin = "<unknown>",
+        TextRowCol textrowcol = {},
+        ImplicitDefault implicit_defaults = ImplicitDefault::YES);
```

Comment:
I feel like this is a layering problem. Just parsing the file shouldn't be doing the semantic transform to figure out what defaults are. Ideally we would have two types: one which is just the parsed guts, and another which is the parsed guts with expressions evaluated and applied.

I understand this suggestion is kind of a large/structural change though :(. @ras0219-msft Would you characterize my statement here as consistent with the dependencies stuff or do you think it's out of left field? (I think it's worth asking autowantwort to go here if this situation is a problem he created but not if we are already filthy about this elsewhere)

Reply:
I have now removed the parameter. 
> just parsing the file shouldn't be doing the semantic transform to figure out what defaults are. 

It is not about what defaults are. It is about if defaults are requested or not. So currently we always use `ImplicitDefault::YES` while parsing.
`liba` => `default-features: true`
`liba[core]` => `default-features: false`
If we add a `ImplicitDefault implicit_defaults` parameter the following would be possible if `ImplicitDefault::NO` is passed:
`liba` => `default-features: false`
But this is semantic currently is never needed (except in one test) so I removed the parameter.  

We could also introduce a new data type for the return value of this function but I currently don't see a value in that. If you really want it I would suggest do not do that in this PR to keep this PR small.